src/app/api/transaction/route.ts

```diff
@@ -0,0 +1,30 @@
+import { Transaction } from '@prisma/client'
+import { NextRequest, NextResponse } from 'next/server'
+
+import prisma from '@/lib/prisma'
+
+export async function POST(req: NextRequest) {
+  const transaction: Transaction = await req.json()
+
+  if (!transaction) {
+    return NextResponse.json(
+      { error: 'Bad Request: Missing Parameters.' },
+      { status: 400 },
+    )
+  }
+
+  try {
+    await prisma.transaction.create({
+      data: transaction,
+    })
+  } catch (e) {
+    return NextResponse.json(
```

Comment:
nit: we should use sentry error logging to detect when a tx fails

also, we should probably add some validation to prevent dos